### ISSUE TITLE
Significant performance regression on latest Master daily builds

### ISSUE THREAD

**Issue by jnativio** [istio/istio#12093]
**Describe the bug**
Regression Patrol has detected that max throughput has been around 20%-25% less than the baseline (1.0.6) for both istio_full and istio_nomixer configurations. Tests were conducted using 1.1.0-rc.0 with similar results. The last 1.1 build with good results was `release-1.1-20190213-09-16`. The next one was below the 10% range, but not as bad as the latest ones, starting with `release-1.1-20190221-09-16`. That was the first build containing the changes of PR #11476. I tried to deploy without those changes and still seeing performance impact. Running again an old build yield good results, what proves it is not an environment problem. 

![image](https://user-images.githubusercontent.com/18554027/53448821-57cfac80-39e6-11e9-8031-c7a049e56b4f.png)

![image](https://user-images.githubusercontent.com/18554027/53448853-6918b900-39e6-11e9-92b4-0b1eb479e4ff.png)

![image](https://user-images.githubusercontent.com/18554027/53448867-746be480-39e6-11e9-8843-f18534f3b071.png)

Complete regression patrol results --> https://ibmcloud-perf.istio.io/regpatrol/

**Expected behavior**
No performance regression

**Steps to reproduce the bug**
Run a workload with 1.1 builds before 02/14 and latest ones and compare the results (throughput and latency)

**Version**
K8s version: 1.10.11_1538
Istio build:  release-1.1-20190221-09-16 and later

**Installation**
Helm template

**Environment**
IKS

@mandarjog

**Comment by mandarjog** [comment]
Is there a change in app? Rquest or limits applied to app?

**Comment by jnativio** [comment]
No change in the app.

**Comment by mandarjog** [comment]
@jnativio can you check  proxy concurrency in both there cases?
```
kubectl --namespace $NS exec -i -t $POD -c istio-proxy -- curl http://localhost:15000/stats | grep concurrency ```

It may be that concurrency has dropped down, which can explain both less cpu and both thruput.

**Comment by jnativio** [comment]
@mandarjog  The concurrency actually went up (from 1 to 2) compared with builds following the merge of #11311.

**Comment by duderino** [comment]
So latency increased but cpu utilization dropped?   That really sounds like concurrency going down.   Did you try running the command that @mandarjog suggested in https://github.com/istio/istio/issues/12093#issuecomment-468495802 ?  That might tell a different story

**Comment by jnativio** [comment]
The istio-proxy concurrency is 2 as it is default since PR #11476. The drop in CPU observed in the Istio node is due to telemetry using less of it, although somehow ingress-gateway is prevented to use more (they run in the same node) and that explains the drop in TPS.

**Comment by jnativio** [comment]
After investigation, I concluded this problem was a composition of 2 causes: 
1. Some change  was affecting the sidecars performance and it was reverted after build `release-1.1-20190227-09-16` on the 1.1 branch but still present on the master branch and requires further investigation. After it was fixed on 1.1, the results for the no-mixer configuration were normal but not for the full Istio, what brings up the second cause.

2. In the regression patrol, all Istio components run in the same node. The recent changes in the default helm values, in particular the `cpu requests`, caused an unbalance between 
the ingress-gateway and telemetry pods when competing for CPU cycles. It appears that the `requests` parameter not just guarantees a chunk of CPU but also works as a weight when pods are competing for extra CPU cycles. The  `cpu requests` default values for telemetry was 1100m per pod (1000m for mixer container plus 100m for the istio-proxy) while for ingress-gateway was left as 10m, which lead to "unfair" constraining the throughput. This is being partially addressed by PR #12315. Regression patrol will change the default values to adequate ones for the restricted environment it runs.
